### DISCUSSION TITLE
404 when deleting a site via API

### DISCUSSION THREAD

**Discussion by s-kris** [plausible/analytics#1695] [Self-Hosted Support] [answered]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [ ] This is a bug report and not a feature request, nor asking for self-hosted support

### Describe the bug

I'm trying to delete a site via API. `404 Not found`error is thrown. This happens on the self-hosted version.

P.S:
Site is created successfully via API.

### Expected behavior

The site has to be deleted via API.

### Screenshots

![Screenshot 2022-02-19 at 9 33 26 AM](https://user-images.githubusercontent.com/3975305/154785622-a61438ee-3091-4c4d-96d2-63f4e8a784d9.png)

### Environment

```markdown
- OS: MacOS
- Browser: Chrome
- Browser Version: 98.0.4758.102

Tested on postman as well.
```

**Comment by s-kris** [comment]
Anyone else facing issue when deleting a site via API?

**Reply by sanjeevkumar761** [reply]
How did you create new site with API? I am struggling to get API key to create new site. Can you please share details about the steps you followed? Thanks a lot.

**Reply by johnxgaehner** [reply]
same here, facing the same problem while trying to delete a site via api. did you find a solution, @s-kris?

**Reply by s-kris** [reply]
@johnxgaehner no, not yet!

**Reply by ricmp** [reply]
same here. I was able to delete the site with API when submitting some form data with it. example:
curl -X DELETE https://localhost:8000/api/v1/sites/test-domain.com \
  -H "Authorization: Bearer ${TOKEN}" \
  -F 'site_id="test-domain.com"'

**Reply by ricmp** [reply]
I've changed my last msg to be more accurate.. it's not POST that you need to send but DELETE. try now, should work.

**Reply by s-kris** [reply]
Thanks for the update. But I still get the same earlier. 
I'm using `plausible/analytics:latest` tag

Can I know which version are you using?

**Reply by ricmp** [reply]
Hi. I am no expert in docker or coding. But I am downloading the fresh version manually from https://github.com/plausible/analytics (master) into the plausible folder under the plausible-hosting, than building it in docker-compose.yml with:
    build:
      context: plausible
      dockerfile: Dockerfile

give that a try and let me know. If that is not working than it's a little code modification I did that got cached and I forgot about it. 

Also, make sure you have this added
https://github.com/plausible/analytics/discussions/1460

**Reply by s-kris** [reply]
It makes sense. I'm pulling the docker image which is an old release.
Thanks for the clarification :)

btw, yeah, I did add the site API key.

**Comment by s-kris** [comment]
@sanjeevkumar761 

https://github.com/plausible/analytics/discussions/1460

**Comment by s-kris** [comment, marked as answer]
@johnxgaehner we may need to wait until the delete API is added to the opensource release

https://github.com/plausible/analytics/blob/master/CHANGELOG.md

**Reply by ukutaht** [reply]
Yes, planning a release this month

**Reply by s-kris** [reply]
Thank you!

**Comment by mattaharish** [comment]
Hi @ukutaht , I have seen that the docker images of plausible/analytics with tags **master** & **stable** are updated very recently. Can we use those tags for self-hosting?  In this file https://github.com/plausible/hosting/blob/f7682057104671b42e3fff3a106f93a32ea14775/docker-compose.yml#L28, **latest** tag is referred for self-hosting. Can you help us in understanding the difference between these tags? We are planning to use the delete site and goal APIs.

Thank You!

**Reply by ukutaht** [reply]
> Can we use those tags for self-hosting?

I wouldn't recommend it, we use these images for development so they're not meant for self-hosting. For self-hosting, use the `latest` or one of the numbered tags.

**Reply by mattaharish** [reply]
Thanks @ukutaht for confirming! Is there any plan to publish the 'latest' tag with changes related to delete site API?

**Reply by ukutaht** [reply]
Yes I'm planning to cut a release next week

**Reply by callmeteus** [reply]
Still waiting for this release since march and nothing until now 😞

**Reply by metmarkosaric** [reply]
Thanks for sharing @theprometeus! We have limited resources and the self-hosted release depends on the free capacity of our developers. Think of it as a long-term release. We have a goal of two releases per year but cannot guarantee or promise any timeline on the next one. Last version was released on January 26th. We will publish the new one as soon as it's ready, thanks!
